renderdoc/driver/d3d12/d3d12_shader_feedback.cpp

```diff
@@ -284,7 +362,14 @@ static bool AnnotateDXILShader(const DXBC::DXBCContainer *dxbc, uint32_t space,
     }
 
     DXIL::Metadata *resources = editor.GetMetadataByName("dx.resources");
-
+    if(!resources)
+    {
+      DXIL::Metadata tmpResList;
+      tmpResList.children.resize(4);
+      DXIL::Metadata tmpResources;
+      tmpResources.children.push_back(editor.AddMetadata(tmpResList));
+      resources = editor.AddMetadata(tmpResources);
+    }
```

Comment:
This fixes the crash I was seeing during patching when `dx.resources` is missing, but it fails to create a pipeline state because the dx.resources named metadata doesn't exist. You'll need to add a new `AddNamedMetadata` I think here, but otherwise I _think_ it shouldn't be complicated.

Reply:
You are right, strangely enough this only fail on one of my two PCs, adding a AddNamedMetadata does fix the fail.